src/kademlia/dht_storage.cpp

```diff
@@ -532,6 +556,36 @@ namespace
 			if (!peers.empty() && peers.capacity() / peers.size() >= 4u)
 				peers.shrink_to_fit();
 		}
+
+		void refresh_infohashes_sample()
```

Comment:
I was just thinking of ways to use a ``std::vector<sha1_hash>`` instead of a ``std::string`` when putting it into an ``entry`` dictionary. ``entry`` has an ``operator=(span<char onst>)``, so if only there was a way to turn an arbitrary ``std::vector<>`` into a span of bytes, you could still assign it as efficiently.

perhaps something like: ``template <typename T> span<char const> to_span(std::vector<T>& v) { return {reinterpret_cast<char const*>(v.data()), v.size() * sizeof(T)}; }``. or something like that